From 7a961dba2db5861ad0d151c7fd12fb3cd8356a2e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?G=C3=BCnther=20Deschner?= <gd@samba.org>
Date: Wed, 20 Jan 2016 17:44:45 +0100
Subject: [PATCH 1/2] param: add parameter "server multi channel support",
 defaults to off.

Guenther

Pair-Programmed-With: Michael Adam <obnox@samba.org>

Signed-off-by: Guenther Deschner <gd@samba.org>
Signed-off-by: Michael Adam <obnox@samba.org>
---
 .../smbdotconf/protocol/servermultichannelsupport.xml | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)
 create mode 100644 docs-xml/smbdotconf/protocol/servermultichannelsupport.xml

diff --git a/docs-xml/smbdotconf/protocol/servermultichannelsupport.xml b/docs-xml/smbdotconf/protocol/servermultichannelsupport.xml
new file mode 100644
index 0000000..b85bbd3
--- /dev/null
+++ b/docs-xml/smbdotconf/protocol/servermultichannelsupport.xml
@@ -0,0 +1,19 @@
+<samba:parameter name="server multi channel support"
+                 context="G"
+                 type="boolean"
+                 xmlns:samba="http://www.samba.org/samba/DTD/samba-doc">
+<description>
+    <para>This boolean parameter controls whether
+    <citerefentry><refentrytitle>smbd</refentrytitle>
+    <manvolnum>8</manvolnum></citerefentry> will support
+    SMB3 multi-channel.
+    </para>
+    <para>This parameter has been added with version 4.4.</para>
+    <para>
+    Warning: Note that this feature is considered experimental in Samba 4.4.
+    Use it at your own risk: it may result in data corruption.
+    </para>
+</description>
+
+<value type="default">no</value>
+</samba:parameter>
-- 
2.5.0


From 1f63533b92b859e78e924cc21b7c305ecf245cc8 Mon Sep 17 00:00:00 2001
From: Michael Adam <obnox@samba.org>
Date: Tue, 26 Jan 2016 08:16:51 +0100
Subject: [PATCH 2/2] smbd: enable multi-channel if 'server multi channel
 support = yes' in the config

Signed-off-by: Michael Adam <obnox@samba.org>
---
 source3/smbd/smbXsrv_client.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/source3/smbd/smbXsrv_client.c b/source3/smbd/smbXsrv_client.c
index 87cc307..8062593 100644
--- a/source3/smbd/smbXsrv_client.c
+++ b/source3/smbd/smbXsrv_client.c
@@ -490,6 +490,8 @@ NTSTATUS smbXsrv_client_create(TALLOC_CTX *mem_ctx,
 	client->ev_ctx = ev_ctx;
 	client->msg_ctx = msg_ctx;
 
+	client->server_multi_channel_enabled = lp_server_multi_channel_support();
+
 	client->table = talloc_move(client, &table);
 	table = client->table;
 
-- 
2.5.0

